glommio/src/executor/placement/mod.rs

```diff
@@ -347,7 +434,7 @@ impl CpuSetGenerator {
     pub fn next(&mut self) -> CpuIter {
         match self {
             Self::Unbound => CpuIter::Unbound,
-            Self::Fenced(cpus) => CpuIter::from_vec(cpus.as_vec().clone()),
+            Self::Fenced(cpus) => CpuIter::from_vec(cpus.as_vec().into_iter().cloned().collect()),
```

Comment:
See my point above. By implementing `into_iter` directly we'd save some work here.

Reply:
Can you elaborate? I added an owning `into_iter` impl, but it's still going to have to clone isn't it (because self is `&mut`)? 

```
Self::Fenced(cpus) => CpuIter::from_vec(cpus.clone().into_iter().collect())
```

Reply:
I thought `self` was owned here so I should’ve said `iter()` and not `into_iter()`. My point was that `as_vec()` was allocating a vec and we could avoid it. You recent does the trick ✅